src/Core/Compiler/CompilerService.cs

```diff
@@ -98,11 +121,18 @@ public AssemblyInfo BuildEntryPoint(OperationInfo operation, CompilerMetadata me
         /// </summary>
         public AssemblyInfo BuildSnippets(Snippet[] snippets, CompilerMetadata metadatas, QSharpLogger logger, string dllName, string executionTarget = null)
         {
+            string openStatements = string.Join("", AutoOpenNamespaces.Select(ns => $"open {ns};"));
             string WrapInNamespace(Snippet s) =>
-                $"namespace {Snippets.SNIPPETS_NAMESPACE} {{ open Microsoft.Quantum.Intrinsic; open Microsoft.Quantum.Canon; {s.code} }}";
+                $"namespace {Snippets.SNIPPETS_NAMESPACE} {{ {openStatements} {s.code} }}";
```

Comment:
One issue we've run into before is that the line numbers and column offsets in error messages don't match up; as the wrapping code gets a bit longer, that may get worse. As a minor suggestion, would it be possible to set the offset when calling `BuildAssembly`?

Reply:
Yes, looks like this should be a simple enough change.